File: frontend/src/types/evaluation.ts
export interface Evaluation {
  id: string;
  solutionId: string;
  dimension: string;
  score: number;
  details: Record<string, unknown>;
  status: string;
  createdAt: string;
  // 智枢评估维度
  rootCauseCut: boolean;
  originalContradictionResolved: boolean;
  newContradictions: string[];
  functionDeficitsFilled: string[];
  newHarmfulInteractions: string[];
  ifrDistance: string;
  ifrGapDescription: string;
  ifrParametersAchieved: string[];
  overallVerdict: string;
  evolutionAlignment: number;
  alignedLaws: string[];
  misalignedLaws: string[];
  maturity: string;
  confidence: number | null;
}

export interface EvaluationSummary {
  id: string;
  solutionId: string;
  score: number;
  dimension: string;
  status: string;
  evaluatedAt: string;
  overallVerdict: string;
  maturity: string;
  confidence: number | null;
}

File: backend/app/models/evaluation.py
from pydantic import BaseModel


class EvaluationCreate(BaseModel):
    solution_id: int
    dimension: str = "comprehensive"
    score: float = 0
    details: dict = {}


class EvaluationResponse(BaseModel):
    id: str
    solution_id: str
    dimension: str
    score: float
    details: dict
    status: str
    created_at: str
